stats_overlay.js

```diff
@@ -0,0 +1,394 @@
+
+  // shift the output based on the size of the canvas and the elaspsed time
+  this.currentXOffset_ = - Math.max(0,
+      (shaka.util.Clock.now() / 1000) - this.startTimestamp_ - this.width_);
+
+  context.clearRect(0, 0, this.canvas_.width, this.canvas_.height);
+
+  this.drawGraph_(context, stats);
+
+  this.drawBufferLengthHistory_(context);
+
+  this.drawStreamHistory_(context, stats);
+
+  this.drawBandwidth_(context, stats);
+
+  this.drawBufferingHistory_(context, stats);
+
+
+};
+
+
+/**
+ * Draw the bandwidth bands
+ * @private
+ * @param {?Object} context
+ * @param {!shaka.player.Stats} stats
+ */
+shaka.StatsOverlay.prototype.drawGraph_ = function(context, stats) {
+
+  context.beginPath();
+  context.moveTo(0, this.height_ + 0.5);
+  context.lineTo(this.width_, this.height_ + 0.5);
+  context.strokeStyle = '#000';
+  context.lineWidth = 0.5;
+  context.stroke();
+
+  // draw bitrates
+  var bitRateInfo = this.player_.getVideoTracks();
+  context.beginPath();
+  context.strokeStyle = '#999';
+  context.lineWidth = 0.5;
+
+  for (var i = 0; i < bitRateInfo.length; i++)
+  {
+    context.moveTo(0,
+        this.convertBitsToYCoordinate_(bitRateInfo[i].bandwidth));
+    context.lineTo(this.width_,
+        this.convertBitsToYCoordinate_(bitRateInfo[i].bandwidth));
+  }
+
+  context.stroke();
+};
+
+
+/**
+ * Draw a blue square line to indicate the stream being played
+ * @private
+ * @param {?Object} context
+ * @param {!shaka.player.Stats} stats
+ */
+shaka.StatsOverlay.prototype.drawStreamHistory_ = function(context, stats) {
+
+  var py = this.height_;
+  var px = 0;
+
+  context.beginPath();
+  context.strokeStyle = '#00b';
+  context.lineWidth = 0.5;
+  context.moveTo(px, this.height_);
+
+  // graph historical stream changes - square graph
+  for (var i = 0; i < stats.streamHistory.length; ++i) {
+
+    // move to the new x
+    px = this.convertTimestampToXCoordinate_(
+        stats.streamHistory[i].timestamp);
+    context.lineTo(px, py);
+
+    // move to the new y - convert from bytes to bits
+    py = this.convertBitsToYCoordinate_(
+        stats.streamHistory[i].value.videoBandwidth);
+    context.lineTo(px, py);
+  }
+
+  // graph to current time
+  px = this.convertTimestampToXCoordinate_(shaka.util.Clock.now() / 1000);
+  context.lineTo(px, py);
+  context.stroke();
+};
+
+
+/**
+ * Draw the bandwidth history as a red line
+ * @private
+ * @param {?Object} context
+ * @param {!shaka.player.Stats} stats
+ */
+shaka.StatsOverlay.prototype.drawBandwidth_ = function(context, stats) {
+
+  var py = 0;
+  var px = 0;
+
+  context.beginPath();
+  context.strokeStyle = '#F00';
+  context.lineWidth = 0.5;
+  context.moveTo(px, this.height_);
+
+  // graph current bandwidth
+  for (var i = 0; i < stats.bandwidthHistory.length; ++i) {
+
+    px = this.convertTimestampToXCoordinate_(
+        stats.bandwidthHistory[i].timestamp);
+    py = this.convertBitsToYCoordinate_(
```

Comment:
When this value is much larger than the highest bitrate, the graphs gets very hard to read because of the scaling.  (I know it's not common at home, but at work I'm getting bandwidth numbers around 30Mbit.)  Perhaps if the estimated bandwidth is more than 2x the highest bitrate of the stream, we should cap the y values and not scale the graph so much that you can't make out the various bitrates.